src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -60,54 +60,74 @@ public function validate(mixed $entity, Constraint $constraint)
             throw new ConstraintDefinitionException('At least one field has to be specified.');
         }
 
-        if (null === $entity) {
+        if (null === $value) {
             return;
         }
 
-        if (!\is_object($entity)) {
-            throw new UnexpectedValueException($entity, 'object');
+        if (!\is_object($value)) {
+            throw new UnexpectedValueException($value, 'object');
         }
 
+        $entityClass = $value::class;
+
         if ($constraint->em) {
             $em = $this->registry->getManager($constraint->em);
 
             if (!$em) {
                 throw new ConstraintDefinitionException(sprintf('Object manager "%s" does not exist.', $constraint->em));
             }
         } else {
-            $em = $this->registry->getManagerForClass($entity::class);
+            $em = $this->registry->getManagerForClass($constraint->entityClass ?? $entityClass);
 
             if (!$em) {
-                throw new ConstraintDefinitionException(sprintf('Unable to find the object manager associated with an entity of class "%s".', get_debug_type($entity)));
+                throw new ConstraintDefinitionException(sprintf('Unable to find the object manager associated with an entity of class "%s".', get_debug_type($value)));
```

Comment:
This should now be dynamic - in case `$constraint->entityClass` is passed. And I'm not sure we need `get_debug_type()` because, at this point, we know that `$value` is an object. So we could change the argument to `sprintf` to `$constraint->entityClass ?? $entityClass`

Reply:
True, I missed that.